tests/pyoozie/test_builder.py

```diff
@@ -130,7 +119,37 @@ def test_coordinator_submission_xml_with_configuration(username, coord_app_path)
     </configuration>''') == tests.utils.xml_to_dict_unordered(actual)
 
 
-def test_workflow_builder(tmpdir, workflow_builder):
+def test_workflow_builder(tmpdir):
+
+    workflow_builder = builder.WorkflowBuilder(
+        name='descriptive-name'
+    ).add_action(
+        name='extract',
+        action=tags.Shell(exec_command='echo', arguments=["'Extract data from an operational system'"]),
+        action_on_error=tags.Email(to='person@example.com', subject='Error',
+                                   body='A bad thing happened while extracting'),
+        kill_on_error='Failure message on extracting',
+    ).add_action(
+        name='transform',
+        action=tags.Shell(exec_command='echo', arguments=["'Transform data'"]),
+        action_on_error=tags.Email(to='person@example.com', subject='Error',
+                                   body='A bad thing happened while transforming'),
+        kill_on_error='Failure message on transforming',
+        depends_upon=('extract',),
+    ).add_action(
+        name='load',
+        action=tags.Shell(exec_command='echo', arguments=["'Load data into a database'"]),
+        action_on_error=tags.Email(to='person@example.com', subject='Error',
+                                   body='A bad thing happened while loading'),
+        kill_on_error='Failure message on loading',
+        depends_upon=('transform',),
+    ).add_transform(
+        transforms.final_action,
+        action=tags.Email(to='person@example.com', subject='Success', body='ETL succeeded'),
+    ).add_transform(
+        transforms.add_actions_on_error
+    )
+
```

Comment:
Above is what the interface would look like to a user who wanted a serial workflow with notifications on an error.